### PR TITLE
DivxTotal improvements. Resolves #11027

### LINKED ISSUE #11027: [divxtotal] only showing first page of search results
**Please use the search bar** at the top of the page and make sure you are not creating an already submitted issue.
Check closed issues as well, because your issue may have already been fixed.

Please read our [Contributing Guidelines](https://github.com/Jackett/Jackett/blob/master/CONTRIBUTING.md) before submitting your issue to ensure a prompt response to your bug.

### Environment

**OS**: Windows 10

**.Net Runtime**: .Net-Runtime

**.Net Version**: Unknown

**Jackett Version**: Unknown

**Last Working Jackett Version**: Unknown

**Are you using a proxy or VPN?** no

### Description

Add a feature to show all search results in all pages. In divxtotal.se shows 400 torrents and in Jackett shows 15.

### Logged Error Messages

No log.

### PR DESCRIPTION
**Pull request by XYZJR**
#11027 

DivxTotal does not have an API and needs a lot of page loads to get the desired query results. Because of this you might get more than 100 releases per query, because each page can contain multiple links to full seasons, episodes, tv series, etc. which is then parsed into separate releases.
I've moved some code to return as soon as it's obvious that the loaded page won't be parsable into a release.
Still, queries take a very long time to complete.

### CODE REVIEW

File: src/Jackett.Common/Indexers/DivxTotal.cs [hunk 1]
```diff
@@ -104,47 +106,52 @@ public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken
...
 
             // we remove parts from the original query
-            query = ParseQuery(query);
-            var qc = new NameValueCollection { { "s", query.SearchTerm } };
+            newQuery = ParseQuery(newQuery);
+            var qc = new NameValueCollection { { "s", newQuery.SearchTerm } };
 
             var page = 1;
-            var isLastPage = false;
+            AngleSharp.Html.Dom.IHtmlDocument htmlDocument = null;
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/DivxTotal.cs:121]: using AngleSharp.Html.Dom.IHtmlDocument to make it less verbose

File: src/Jackett.Common/Indexers/DivxTotal.cs [hunk 2]
```diff
@@ -154,102 +161,122 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
+        /// <Exception cref="ExceptionWithConfigData" />
+        private async Task<string> LoadWebPageAsync(string url)
+        {
+            var result = await RequestWithCookiesAsync(url);
+            return result.Status == HttpStatusCode.OK
+                ? result.ContentString
+                : throw new ExceptionWithConfigData(result.ContentString, configData);
+        }
+
+        private AngleSharp.Html.Dom.IHtmlDocument ParseHtmlIntoDocument(string htmlContentString)
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/DivxTotal.cs:185]: private IHtmlDocument ParseHtmlIntoDocument(string htmlContentString)

File: src/Jackett.Common/Indexers/DivxTotal.cs [hunk 3]
```diff
@@ -154,102 +161,122 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
+            => new HtmlParser().ParseDocument(htmlContentString);
+
+        private bool IsLastPageOfQueryResult(AngleSharp.Html.Dom.IHtmlDocument htmlDocument)
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/DivxTotal.cs:188]: private bool IsLastPageOfQueryResult(HtmlDocument htmlDocument)

**Review by ngosang — APPROVED** [review]
The code looks good in general.
Thank you!

**Comment by ngosang** [comment]
@ilike2burnthing @garfield69 @XYZJR
I gave @XYZJR write permission in this repository (check your confirmation email). You will be able to do more things like assign yourself to issues, open and close issues, etc...
* Feel free to improve the C# indexers or add new ones. I can still review your code if you want, but I think you have enough expertise. Some indexers have complex logic to handle IMDB searches, remove words, etc. Be careful when you change existing code.
* It's really hard to break Jackett if you only change the indexers code. If you change the "core code" add me as a reviewer. Jackett's code is easy to understand but the code has to work in several OS, architectures, NASs, Docker, Mono... Although we have continuous integration, we have to go slowly. 
* Take care of the code changed by you. I know this is not going to happen with you but in the past some users made important changes in the code without enough testing and they leave the project. I had to spend a lot of time fixing those issues. Be responsible.
* Welcome! Here you have some issues you can work on => https://github.com/Jackett/Jackett/projects/3

**Comment by XYZJR** [comment]
Thanks! I'll keep adding you as a reviewer since it's always better to have an extra pair of eyes on things.

**Review by ngosang — APPROVED** [review]

**Pull request merged**